arclightning_backend/src/router.rs

```diff
@@ -128,18 +155,67 @@ impl Router {
                         .map_err(|err| {
                             io::Error::new(
                                 ErrorKind::Other,
-                                format!("An error occured when building a response: {}", err)
+                                format!("An error occured when building a response: {}", err),
                             )
                         })
                 })
             }).flatten();
         Box::new(response)
     }
 
+    fn serve_static_file(
+        &self,
+        root: PathBuf,
+        valid_files: Vec<PathBuf>,
+        mut request: Request<Body>,
+    ) -> ResponseFuture {
+        // TODO: set up a 404 page. Maybe hyper static file does it?
+
+        let requested_path = &root.join(
+            match PathBuf::from(&request.uri().path()).strip_prefix("/") {
+                // strip_prefix(x) returns an error if the PathBuf does not
+                // start with x. This situation should not occur in the web browser
+                // but in case it does, we just return the path without removing
+                // the initial "/"
+                Ok(v) => v,
+                Err(_err) => Path::new(request.uri().path()),
+            },
+        );
+
+        if requested_path == &root || requested_path == &root.join(PathBuf::from(r"\\")) {
```

Comment:
what is this `\\` thing

Reply:
Due to how we're storing all valid filepaths for serving with `PathBuf`, all `/` get converted to `\\` for the Windows OS.
Specifically, `requested_path` is a `PathBuf` that uses OS-specific delimiters.  In order to compare to the URLs we want to associate with `index.html`, I need to hard code the Windows delimiter.

Reply:
hmm. I would prefer to try to handle this using pathbuf functions, since using strings to handle paths is kinda icky